### PR TITLE
add tests on be

### PR DESCRIPTION
**Pull request by RevuraV**

### CODE REVIEW

File: server/src/controllers/like.controllers/tests/add.like.test.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { addLike } from '../add.like';
+import { request, response } from 'express';
+import mockingoose from 'mockingoose';
+import * as likeRequests from '../../../db.requests/add.like.requests';
+import { Like, ILikeModel } from '../../../models/like.model';
+
+type IResolve<T> = (value: T) => void;
+
+const fakeNext = jest.fn(() => { /* */
+});
```
Comment (anna-kuzii) [inline server/src/controllers/like.controllers/tests/add.like.test.ts]: one line

File: server/src/controllers/feed.controllers/tests/get.feed.test.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { getFeed } from '../get.feed';
+import { request, response } from 'express';
+import mockingoose from 'mockingoose';
+import * as getRequests from '../../../db.requests/getFeed.requests';
+import { Post } from '../../../models/post.model';
+
+type IResolve<T> = (value: T) => void;
+
+const fakeNext = jest.fn(() => { /* */
```
Comment (anna-kuzii) [inline server/src/controllers/feed.controllers/tests/get.feed.test.ts]: one line

File: server/src/controllers/feed.controllers/tests/get.feed.test.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+});
+
+describe('Get feed controller:', () => {
+    test('get feed - success', async () => {
+        mockingoose(Post).toReturn({}, 'findOne');
+        const fakeFeed: object[] = await Post.find({}) as object[];
+
+        const mockCountLike = jest.spyOn(getRequests, 'getPostsForFeed');
+        const value1 = new Promise((res: IResolve<object[]>): void => res(fakeFeed));
```
Comment (anna-kuzii) [inline server/src/controllers/feed.controllers/tests/get.feed.test.ts]: bad name

File: server/src/controllers/feed.controllers/tests/get.feed.test.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        mockCountLike.mockReturnValue(value1);
+
+        request.params = {
+            page: 1,
+        };
+        response.json = jest.fn(() => response);
+
+        await getFeed(request, response, fakeNext);
+        expect(response.json).toHaveBeenCalledTimes(0);
```
Comment (anna-kuzii) [inline server/src/controllers/feed.controllers/tests/get.feed.test.ts]: why 0?

File: server/src/controllers/like.controllers/tests/get.likes.test.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import { getLikes } from '../get.likes';
+import { request, response } from 'express';
+import mockingoose from 'mockingoose';
+import * as likeRequests from '../../../db.requests/get.likes';
+import { Like, ILikeModel } from '../../../models/like.model';
+
+type IResolve<T> = (value: T) => void;
+
+const fakeNext = jest.fn(() => { /* */
```
Comment (anna-kuzii) [inline server/src/controllers/like.controllers/tests/get.likes.test.ts]: one line

File: server/src/controllers/like.controllers/tests/remove.like.test.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
+import { removeLike } from '../remove.like';
+import { request, response } from 'express';
+import mockingoose from 'mockingoose';
+import * as likeRequests from '../../../db.requests/remove.like.requsets';
+import { Like, ILikeModel } from '../../../models/like.model';
+
+type IResolve<T> = (value: T) => void;
+
+const fakeNext = jest.fn(() => { /* */
+});
```
Comment (anna-kuzii) [inline server/src/controllers/like.controllers/tests/remove.like.test.ts]: u know

File: server/src/controllers/post.controllers/tests/update.post.by.id.test.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
+import { update } from '../update.post.by.id';
+import { request, response } from 'express';
+import mockingoose from 'mockingoose';
+import * as postRequests from '../../../db.requests/updatePost.request';
+import { Post, IPostModel } from '../../../models/post.model';
+
+type IResolve<T> = (value: T) => void;
+
+const fakeNext = jest.fn(() => { /* */
+});
```
Comment (anna-kuzii) [inline server/src/controllers/post.controllers/tests/update.post.by.id.test.ts]: !

File: server/src/controllers/post.controllers/tests/update.post.by.id.test.ts [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+describe('Update post controller:', () => {
+    test('update post - success', async () => {
+        mockingoose(Post).toReturn({}, 'findOne');
+        const fakePost: IPostModel = await Post.findOne({}) as IPostModel;
+
+        const mockUpdatePost = jest.spyOn(postRequests, 'updatePost');
+        const value1 = new Promise((res: IResolve<IPostModel>): void => res(fakePost));
```
Comment (anna-kuzii) [inline server/src/controllers/post.controllers/tests/update.post.by.id.test.ts]: !

**Review by anna-kuzii — CHANGES_REQUESTED** [review]

File: server/src/controllers/like.controllers/tests/remove.like.test.ts (hunk 6 above)
Comment (RevuraV) [reply server/src/controllers/like.controllers/tests/remove.like.test.ts]: fixed

File: server/src/controllers/post.controllers/tests/update.post.by.id.test.ts (hunk 7 above)
Comment (RevuraV) [reply server/src/controllers/post.controllers/tests/update.post.by.id.test.ts]: fixed

File: server/src/controllers/post.controllers/tests/update.post.by.id.test.ts (hunk 8 above)
Comment (RevuraV) [reply server/src/controllers/post.controllers/tests/update.post.by.id.test.ts]: fixed

File: server/src/controllers/like.controllers/tests/get.likes.test.ts (hunk 5 above)
Comment (RevuraV) [reply server/src/controllers/like.controllers/tests/get.likes.test.ts]: fixed

File: server/src/controllers/feed.controllers/tests/get.feed.test.ts (hunk 4 above)
Comment (RevuraV) [reply server/src/controllers/feed.controllers/tests/get.feed.test.ts]: some tests want 0

File: server/src/controllers/feed.controllers/tests/get.feed.test.ts (hunk 3 above)
Comment (RevuraV) [reply server/src/controllers/feed.controllers/tests/get.feed.test.ts]: fixed

File: server/src/controllers/feed.controllers/tests/get.feed.test.ts (hunk 2 above)
Comment (RevuraV) [reply server/src/controllers/feed.controllers/tests/get.feed.test.ts]: fixed

File: server/src/controllers/like.controllers/tests/add.like.test.ts (hunk 1 above)
Comment (RevuraV) [reply server/src/controllers/like.controllers/tests/add.like.test.ts]: fixed

File: server/src/controllers/feed.controllers/tests/get.feed.test.ts (hunk 4 above)
Comment (RevuraV) [reply server/src/controllers/feed.controllers/tests/get.feed.test.ts]: but i fixed this

**Review by anna-kuzii — APPROVED** [review]

**Review by IhorKalyta — APPROVED** [review]

**Pull request merged**